File: 1469-minimum-number-of-steps-to-make-two-strings-anagram/minimum-number-of-steps-to-make-two-strings-anagram.kt
class Solution {
    fun minSteps(s: String, t: String): Int {
        val frequencies = s.asSequence()
            .groupingBy { it }
            .eachCount()
            .toMutableMap()

        t.forEach {
            val f = frequencies.getOrDefault(it, 0)
            if (f > 0) {
                frequencies[it] = f - 1
            }
        }
        return frequencies.values.sumOf { it }
    }
}

File: 1469-minimum-number-of-steps-to-make-two-strings-anagram/minimum-number-of-steps-to-make-two-strings-anagram.swift
class Solution {
    func minSteps(_ s: String, _ t: String) -> Int {
var hash = [Character: Int]()
    
    for i in t {
        if let map = hash[i] {
            hash[i] = map + 1
        } else {
            hash[i] = 1
        }
    }
    
    var changes = 0
    for i in s {
        if let map = hash[i] {
            let newSum = map - 1
            hash[i] = newSum > 0 ? newSum : nil
        } else {
            changes += 1
        }
    }
    
    return changes

    }
}